app/View/Services/frmServicesModal.cs

```diff
@@ -31,26 +29,88 @@ private void FrmServicesModal_Load(object sender, EventArgs e)
+                    MessageBox.Show("Description and Price are required.");
+                    return;
+                }
+
+                // Check if the price text contains any spaces
+                if (txtPrice.Text.Contains(" "))
+                {
+                    MessageBox.Show("Price must not contain any spaces.");
+                    return;
+                }
+
+                // Check for leading zeros (excluding valid "0" case)
+                if (txtPrice.Text.StartsWith("0") && txtPrice.Text.Length > 1 && !txtPrice.Text.StartsWith("0."))
+                {
+                    MessageBox.Show("Price cannot have leading zeros. Please enter a valid positive number.");
+                    return;
+                }
+
+                // Try to parse the price as a decimal
+                if (!decimal.TryParse(txtPrice.Text, out decimal price))
+                {
+                    MessageBox.Show("Price must be a valid number.");
+                    return;
+                }
+
+                // Check if the price is negative
+                if (price < 0)
+                {
+                    MessageBox.Show("Price cannot be negative. Please enter a valid positive number.");
+                    return;
+                }
+
+                // Create a service instance with values
+                var service = new app.core.model.Services
+                {
+                    Id = this.Id, // Assuming this.Id will be 0 for new services
+                    ServiceCode = txtCode.Text.Trim(), // Trim to remove extra spaces
+                    Description = txtDesc.Text.Trim(),
+                    Price = price.ToString("F2") // Format price to 2 decimal places
                 };
 
-                ServiceRepository serviceRepository = new ServiceRepository();
-                if (serviceRepository.SaveService(services))
+                // Use repository to check for duplicate service code
+                var serviceRepository = new ServiceRepository();
+                bool isDuplicate = service.Id == 0
+                    ? serviceRepository.IsDuplicateServiceCode(service.ServiceCode, 0) // Check for new service
+                    : serviceRepository.IsDuplicateServiceCode(service.ServiceCode, service.Id); // Check for updates
+
+                if (isDuplicate)
                 {
-                    MessageBox.Show("Saved successfully");
+                    MessageBox.Show("The service code already exists. Please use a different code.", "Duplicate Service Code", MessageBoxButtons.OK, MessageBoxIcon.Warning);
+                    return;
+                }
+
+                // Save or update the service
+                bool success = serviceRepository.SaveService(service);
+                if (success)
+                {
+                    MessageBox.Show(service.Id == 0 ? "Saved successfully." : "Service updated successfully.");
+
+                    LoadServiceDetails();
```

Comment:
During adding new service record, this code block will throw an error due to the default value of `Id`=0.
![image](https://github.com/user-attachments/assets/5be97d44-da82-4a8c-82b5-8af93fdb58b3)

I believe this method `LoadServiceDetails()` is only used to fetch and populate the fields in the ServiceModal during update.

I don't see any significance why do we need to call it here, since the record was already added/updated in any case.

```suggestion
        // Remove the code including LoadServiceDetails(); in Line 96 to 97.
```